tests/testMatchPessimisticB.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, division, print_function
+
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import math
+import os
+import unittest
+
+import lsst.daf.base as dafBase
+import lsst.pipe.base as pipeBase
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+import lsst.utils.tests
+import lsst.afw.image as afwImage
+from lsst.meas.algorithms import LoadReferenceObjectsTask
+import lsst.meas.astrom.sip.genDistortedImage as distort
+import lsst.meas.astrom as measAstrom
+
+
+class TestMatchOptimisticB(unittest.TestCase):
+
+    def setUp(self):
+
+        self.config = measAstrom.MatchPessimisticBTask.ConfigClass()
+        self.MatchPessimisticB = measAstrom.MatchPessimisticBTask(config=self.config)
+
+        metadata = dafBase.PropertySet()
+        metadata.set("RADECSYS", "FK5")
+        metadata.set("EQUINOX", 2000.0)
+        metadata.set("CTYPE1", "RA---TAN")
+        metadata.set("CTYPE2", "DEC--TAN")
+        metadata.set("CUNIT1", "deg")
+        metadata.set("CUNIT2", "deg")
+        metadata.set("CRVAL1", 36.930640)
+        metadata.set("CRVAL2", -4.939560)
+        metadata.set("CRPIX1", 792.4)
+        metadata.set("CRPIX2", 560.7)
+        metadata.set("CD1_1", -5.17e-05)
+        metadata.set("CD1_2", 0.0)
+        metadata.set("CD2_2", 5.17e-05)
+        metadata.set("CD2_1", 0.0)
+        self.wcs = afwImage.makeWcs(metadata)
+        self.distortedWcs = self.wcs
```

Comment:
self.distoredWcs is a copy at this point but if you look at the later tests, which are the same as testMatchOptimisticB it does end up being set to a distorted wcs in testLargeDistortion and for the individual distort tests (e.g. testLinearXDistort). PS I didn't add this test, it must be showing up incorrectly in the commit history.

This sounds like another ticket to make a better test for both MatchOptimistic and PessimsiticB. I'm happy to collaborate and work on it. I'm not going to be as familiar with all of the cases we want to test for optical distortion.